packages/runner/lib/server.ts

```diff
@@ -60,31 +57,28 @@ function startProcedure() {
                 fileLocation: nangoProps.syncConfig.file_location,
                 input: codeParams
             });
-            usage.track(nangoProps);
-            // executing in the background and returning immediately
-            // sending the result to the jobs service when done
-            setImmediate(async () => {
-                const heartbeat = setInterval(async () => {
-                    await jobsClient.postHeartbeat({ taskId });
-                }, heartbeatIntervalMs);
-                try {
-                    const abortController = new AbortController();
-                    abortControllers.set(taskId, abortController);
-
-                    const { error, response: output } = await exec({ nangoProps, code, codeParams, abortController, locks });
-
-                    await jobsClient.putTask({
-                        taskId,
-                        nangoProps,
-                        ...(error ? { error } : { output: output as any })
-                    });
-                } finally {
-                    clearInterval(heartbeat);
-                    abortControllers.delete(taskId);
-                    usage.untrack(nangoProps);
-                    logger.info(`Task ${taskId} completed`);
+            const worker = new RunnerWorker({
+                taskId,
+                jobsServiceUrl,
+                heartbeatIntervalMs: envs.RUNNER_HEARTBEAT_INTERVAL_MS,
+                memoryCheckIntervalMs: envs.RUNNER_MEMORY_CHECK_INTERVAL_MS,
+                locksBuffer: locks.getBuffer(),
+                nangoProps,
+                code,
+                codeParams
+            });
+            worker.start();
+            worker.on('error', (err) => {
+                logger.error(`Task ${taskId} failed`, { err });
+            });
+            worker.on('exit', (exitCode) => {
+                if (exitCode) {
+                    logger.error(`Task ${taskId} exited with code ${exitCode}`);
                 }
+                monitor.untrack(worker);
             });
+            monitor.track(worker);
```

Comment:
Can you track it just before start, just to be sure?

Reply:
yes